### PR TITLE
SimpleSpanProcessor without sampling 

### PR DESCRIPTION
**Pull request by jeffreye**
https://github.com/open-telemetry/opentelemetry-java/issues/4467
https://github.com/open-telemetry/opentelemetry-specification/issues/2543

### CODE REVIEW

File: sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/Sampler.java [hunk 1]
```diff
@@ -89,6 +89,25 @@ static Sampler traceIdRatioBased(double ratio) {
...
+   * are sampled can be relied on, not how the sampled traces are determined. As such, it is
+   * recommended to only use this {@link Sampler} for root spans using {@link
+   * Sampler#parentBased(Sampler)}.
+   *
+   * @param ratio The desired ratio of sampling. Must be within [0.0, 1.0].
+   * @param negativeSamplingResult The sampling result when it's a negative decision
+   * @return a new TraceIdRatioBased {@link Sampler}.
+   * @throws IllegalArgumentException if {@code ratio} is out of range
+   */
+  static Sampler traceIdRatioBased(double ratio, SamplingResult negativeSamplingResult) {
```
Comment (anuraaga) [inline sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/Sampler.java:107]: Hi @jeffreye - we originally had a knob for exporting spans regardless of sampling decision but removed it not seeing a use case for it (but forgot to remove the logic related to it). I think I follow your use case, but it seems to be predicated on this sort of sampling strategy - we would need to add it in the specification before adding it here though. Can you file an issue at https://github.com/open-telemetry/opentelemetry-specification and change this PR to draft pending spec for this?

Comment (jeffreye) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/Sampler.java:107]: I just realized it's a spec change too. Submitted here https://github.com/open-telemetry/opentelemetry-specification/issues/2543

Comment (jeffreye) [reply sdk/trace/src/main/java/io/opentelemetry/sdk/trace/samplers/Sampler.java:107]: Just out of curious, was `SamplingDecision.RECORD_ONLY` designed for this?

**Comment by jeffreye** [comment]
on hold until https://github.com/open-telemetry/opentelemetry-specification/issues/2543

**Pull request closed without merging**